CMakeLists.txt

```diff
@@ -0,0 +1,455 @@
+  ${CMAKE_CURRENT_SOURCE_DIR}/fdbclient/vexillographer/ruby.cs
+  ${CMAKE_CURRENT_SOURCE_DIR}/fdbclient/vexillographer/vexillographer.cs)
+
+set(VEXILLOGRAPHER_REFERENCES "-r:System,System.Core,System.Data,System.Xml,System.Xml.Linq")
+set(VEXILLOGRAPHER_EXE "${CMAKE_CURRENT_BINARY_DIR}/vexillographer.exe")
+add_custom_command(OUTPUT ${VEXILLOGRAPHER_EXE}
+  COMMAND ${MCS_EXECUTABLE} ARGS ${VEXILLOGRAPHER_REFERENCES} ${VEXILLOGRAPHER_SRCS} -target:exe -out:${VEXILLOGRAPHER_EXE}
+  DEPENDS ${VEXILLOGRAPHER_SRCS}
+  COMMENT "Compile Vexillographer")
+add_custom_target(vexillographer DEPENDS ${VEXILLOGRAPHER_EXE})
+
+set(ERROR_GEN_SRCS
+  ${CMAKE_CURRENT_SOURCE_DIR}/flow/error_gen.cs)
+set(ERROR_GEN_REFERENCES "-r:System,System.Core,System.Data,System.Xml,System.Xml.Linq")
+set(ERROR_GEN_EXE "${CMAKE_CURRENT_BINARY_DIR}/error_gen.exe")
+add_custom_command (OUTPUT ${ERROR_GEN_EXE}
+  COMMAND ${MCS_EXECUTABLE} ARGS ${ERROR_GEN_REFERENCES} ${ERROR_GEN_SRCS} -target:exe -out:${ERROR_GEN_EXE}
+  DEPENDS ${ERROR_GEN_SRCS}
+  COMMENT "Compile error_gen")
+add_custom_target(error_gen DEPENDS ${ERROR_GEN_EXE})
+
+################################################################################
+# Helper functions
+################################################################################
+
+if (OPEN_FOR_IDE)
+  macro(actor_set varname srcs)
+    set(${varname})
+    foreach(src ${srcs})
+	    set(tmp "${src}")
+	    set(${varname} "${${varname}};${tmp}")
+    endforeach()
+  endmacro()
+
+  macro(actor_compile target srcs)
+  endmacro()
+else()
+  macro(actor_set varname srcs)
+    set(${varname})
+    foreach(src ${srcs})
+	    set(tmp "${src}")
+	    if(${src} MATCHES ".*\\.h")
+	      continue()
+	    elseif(${src} MATCHES ".*\\.actor\\.cpp")
+	      string(REPLACE ".actor.cpp" ".actor.g.cpp" tmp ${src})
+	      set(tmp "${CMAKE_CURRENT_BINARY_DIR}/${tmp}")
+	    endif()
+	    set(${varname} "${${varname}};${tmp}")
+    endforeach()
+  endmacro()
+
+  set(ACTOR_TARGET_COUNTER "0")
+  macro(actor_compile target srcs)
+    set(options DISABLE_ACTOR_WITHOUT_WAIT)
+    set(oneValueArg)
+    set(multiValueArgs)
+    cmake_parse_arguments(ACTOR_COMPILE "${options}" "${oneValueArgs}" "${multiValueArgs}" "${ARGN}")
+    set(_tmp_out "")
+    foreach(src ${srcs})
+	    set(tmp "")
+	    if(${src} MATCHES ".*\\.actor\\.h")
+	      string(REPLACE ".actor.h" ".actor.g.h" tmp ${src})
+	    elseif(${src} MATCHES ".*\\.actor\\.cpp")
+	      string(REPLACE ".actor.cpp" ".actor.g.cpp" tmp ${src})
+	    endif()
+	    set(actor_compiler_flags "")
+      if(ACTOR_COMPILE_DISABLE_ACTOR_WITHOUT_WAIT)
+        set(actor_compiler_flags "--disable-actor-without-wait-error")
+      endif()
+	    if(tmp)
+	      add_custom_command(OUTPUT "${CMAKE_CURRENT_BINARY_DIR}/${tmp}"
+		      COMMAND ${MONO_EXECUTABLE} ${actor_exe} "${CMAKE_CURRENT_SOURCE_DIR}/${src}" "${CMAKE_CURRENT_BINARY_DIR}/${tmp}" ${actor_compiler_flags} > /dev/null
```

Comment:
I will move this to a different file for now. I was not aware that there are layers that are written in flow. Long term it might make sense to make an install-target `flow` which will also install this cmake file.